app/src/main/java/com/example/marvel_app/BindingAdapter.kt

```diff
@@ -0,0 +1,68 @@
+package com.example.marvel_app
+
+import android.graphics.drawable.Drawable
+import android.util.Log
+import android.view.View
+import android.widget.ImageView
+import android.widget.TextView
+import androidx.core.net.toUri
+import androidx.databinding.BindingAdapter
+import androidx.recyclerview.widget.RecyclerView
+import com.bumptech.glide.Glide
+import com.bumptech.glide.load.DataSource
+import com.bumptech.glide.load.engine.GlideException
+import com.bumptech.glide.request.RequestListener
+import com.bumptech.glide.request.target.Target
+import com.example.marvel_app.model.Comic
+
+@BindingAdapter("comics")
+fun bindRecyclerView(recyclerView: RecyclerView, comics: List<Comic>?){
+    val adapter = recyclerView.adapter as ComicsAdapter
+    adapter.submitList(comics)
+}
+
+@BindingAdapter("authors")
+fun formatAuthors(textView: TextView, authors: List<String>?){
+    if (authors != null) {
+        var formattedText = "written by "
+        for (author in authors) {
+            formattedText += "$author, "
+        }
+        formattedText = formattedText.subSequence(0, formattedText.length - 2) as String
+        textView.text = formattedText
+    }
+}
+
+@BindingAdapter("image_url")
+fun bindImage(imgView: ImageView, imageUrl: String?) {
+    if (imageUrl != null) {
```

Comment:
It can be replaced to use kotlin syntax:
imageIrl?.let { what to do if not null } ?: run { what to do if null }